numpy/_core/tests/test_mem_overlap.py

```diff
@@ -643,16 +643,15 @@ def check_unary_fuzz(self, operation, get_out_axis_size, dtype=np.int16,
                                 sl = [slice(0, 1)] + [0] * (ndim - 1)
                             else:
                                 sl = [slice(0, outsize)] + [0] * (ndim - 1)
-                        else:
-                            if outsize is None:
-                                k = b.shape[axis] // 2
-                                if ndim == 1:
-                                    sl[axis] = slice(k, k + 1)
-                                else:
-                                    sl[axis] = k
+                        elif outsize is None:
+                            k = b.shape[axis] // 2
+                            if ndim == 1:
+                                sl[axis] = slice(k, k + 1)
                             else:
-                                assert b.shape[axis] >= outsize
-                                sl[axis] = slice(0, outsize)
+                                sl[axis] = k
```

Comment:
I believe there's some other ruff rule that wants us to rewrite this as 

```py
sl[axis] = slice(k, k + 1) if ndim == 1 else k
```

but it's fine by me to postpone it until it's enabled

Reply:
I think that would be rule [`SIM108`](https://docs.astral.sh/ruff/rules/if-else-block-instead-of-if-exp/), but many maintainers do not like `SIM` rules because they find the resulting code less readable.